docs/SETTINGS.md

```diff
@@ -0,0 +1,195 @@
+# Repository Settings
+
+---
+
+These are the default settings that should be configured in each repo, changes can be made as needed.
+The settings described here should serve as a baseline, it is more important that there is some setting in place rather 
+than the exact settings being used across every repo.
+
+After a repo's settings are configured this file can be safely deleted from the repo since it is not needed for the
+project to run.
+
+## General
+
+---
+
+### Default Branch
+- Default branch should be `main`
+
+### Features
+- Enable `Issues` 
+- Enable `Projects` 
+
+`Wikis` and `Discussions` can be enabled if needed.
+
+### Pull Requests
+- Disable allow merge commits
+- Enable squash merging with default commit message set to `Pull request title`
+- Disable allow rebase merging
+- Enable always suggest updating pull request branches
+- Enable allow auto-merge
+- Enable automatically delete head branches
+
+## Collaborators and teams  
+
+---
+
+Every repo should have a team or set of teams configured in the `Manage Access` section. Ideally there should be
+two teams where one team is given `Admin` access and the other team is given either `Maintain` or `Write` access.
+
+The team responsible for code review should also be added to the [CODEOWNERS](../.github/CODEOWNERS) file so that
+they are automatically added as reviewers when a PR is opened. Read more on code owners [here](https://help.github.com/articles/about-codeowners/).
+
+## Moderation options
+
+---
+
+### Interaction limits
+
+Default settings should be alright here and can also be configured across the entire organization 
+in the [organization settings](https://github.com/organizations/strangelove-ventures/settings/interaction_limits).
+
+### Code review limits
+
+Default settings should be alright here.
+
+### Reported content
+
+Default settings should be alright here.
+
+## Branches
+
+---
+
+A new branch protection rule should be created for the `main` branch. This rule should have the following settings:
+
+- Enable require a pull request before merging
+- Enable require approvals with at least 1 approval needed before merging
+- Enable require review from Code Owners
+- Enable require status checks to pass before merging
+  - TODO: Fill out this section with all status checks that should be required
+  - This will have a list of all CI actions that MUST pass in order for a PR to be merged.
+  - A conversation should be had with the team to determine what these checks should be before accepting this doc.
```

Comment:
This PR should not be merged until this section is finished. We should configure this section with each CI action that is required to pass before a PR can be merged into main.

I personally think that every CI action we have configured should be required to pass before new code can be merged into main or any other protected branches. This would include,

- PR titles follow conventional commit standard
- Spell checker detects no spelling mistakes
- No markdown links are broken
- No linter warnings/errors are detected
- All unit, integration and e2e tests are passing
- CodeQL analysis does not detect any code smells, security vulnerabilities, etc.
- Compiling the codebase from the branch used in the PR is successful 

I would love to hear others' thoughts on this though since currently I believe some of these things are not required to pass in our existing repos (e.g. i don't believe we enforce some things like the spell checker, broken link detector, PR titles following conventional commits, etc.)

cc @joelsmith-2019 @Reecepbcups @beckettsean

Reply:
I am a no on requiring CodeQL for merging. I did this at Juno & with how cosmos handles security issues, it can block us for a while with no actual patch (happened with the SDK 2 times). They released patches for versions not tagged for another few weeks. If that can be seperated from code smells and other things, I am for it

Will review this in more detail tomorrow

Reply:
Let's start with things as warnings to begin and we can promote them to blockers later when we have confidence in the pattern.

For example, the markdown link checker already broke for us when trying to add relative links while on a branch. That seems like a common-enough pattern that we may not want to require that it pass.

However, CodeQL and Markdown link checks should absolutely be performed for every PR, and we can (for now) manually let folks know as part of the review that they must pass before we will merge the PR. For the cases where they are erroneous or irrelevant, we can still merge the PR without them passing.

Reply:
To be explicit with my opinion (incorporating Reece's feedback)

## Required to merge
- PR titles follow conventional commit standard
- Spell checker detects no spelling mistakes
- No linter warnings/errors are detected
- All unit, integration and e2e tests are passing
- Compiling the codebase from the branch used in the PR is successful

## Tested with every PR but can merge even if they are failing
- No markdown links are broken
- CodeQL analysis does not detect any code smells, security vulnerabilities, etc.

Reply:
This sounds good to me!

Reply:
Added this in e83a56bf29371cbde4fba038ab29724762ef2d6e